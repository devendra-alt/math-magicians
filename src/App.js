import Calculator from './components/calculator';
import './styles/App.css';

function App() {
  return <Calculator />;
}

export default App;
