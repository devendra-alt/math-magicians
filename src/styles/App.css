/* calculator */
:root {
  --calculator-screen-primary: #e0e0e0;
  --calculator-screen-ans-primary: #858694;
  --calculator-screen-last-row: #f5913e;
  --primary-white: #fff;
  --primary-black: #000;
}

.calculator-container {
  display: flex;
}

table {
  margin: 0 auto;
}

td {
  width: 10rem;
  height: 5.5rem;
  text-align: center;
  background-color: var(--calculator-screen-primary);
  font-size: 1.4rem;
  font-weight: 500;
  cursor: pointer;
}

td:last-child {
  background-color: var(--calculator-screen-last-row);
}

#calculator-answer-el {
  background-color: var(--calculator-screen-ans-primary);
  text-align: right;
  padding-right: 2.5rem;
  font-size: 2rem;
  font-weight: 700;
  color: var(--primary-white);
  cursor: auto;
}

::selection {
  color: #000;
}

@media (min-width: 764px) {
  .calculator-container {
    justify-content: flex-end;
    padding: 8rem 16rem;
  }

  table {
    margin: 0;
  }
}
