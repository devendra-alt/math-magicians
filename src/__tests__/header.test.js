import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import {
  render,
  screen,
  waitFor,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import Header from '../components/header';
import App from '../App';
describe('renders correctly', () => {
  beforeEach(() => {
    render(<App />);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render header on screen', () => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );
    waitFor(() =>
      expect(screen.findByText('Math Magicians')).toBeInTheDocument()
    );
  });

  it('should navigate to calculator', async () => {
    const user = userEvent.setup();
    await user.click(screen.getByText(/calculator/i));
    expect(screen.getByText(/Let's do some math/i)).toBeInTheDocument();
  });
  it('should naviaget to quotes', async () => {
    const user = userEvent.setup();
    await user.click(screen.getByText(/Quotes/i));
    await waitForElementToBeRemoved(screen.getByLabelText('loading-msg'));
    expect(screen.getByRole('button')).toBeInTheDocument();
  });
});
