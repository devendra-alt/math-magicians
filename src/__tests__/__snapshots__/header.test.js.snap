// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly renders correctly 1`] = `
<header
  className="header"
>
  <nav
    className="nav-bar"
  >
    <h1
      className="title"
    >
      Math Magicians
    </h1>
    <ul
      className="nav-links"
    >
      <li
        className="nav-link-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-link-item"
      >
        <a
          className="nav-link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="nav-link-item"
      >
        <a
          className="nav-link"
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
